src/servers/sapphire_zone/Actor/Player.cpp

```diff
@@ -159,43 +169,32 @@ bool Core::Entity::Player::isMarkedForRemoval() const
 
 Core::Common::OnlineStatus Core::Entity::Player::getOnlineStatus()
 {
-   uint64_t newMask = uint64_t( 1 ) << static_cast< uint32_t >( OnlineStatus::NewAdventurer );
-   uint64_t afkMask = uint64_t( 1 ) << static_cast< uint32_t >( OnlineStatus::AwayfromKeyboard );
-   uint64_t busyMask = uint64_t( 1 ) << static_cast< uint32_t >( OnlineStatus::Busy );
-   uint64_t dcMask = uint64_t( 1 ) << static_cast< uint32_t >( OnlineStatus::Disconnected );
-   uint64_t meldMask = uint64_t( 1 ) << static_cast< uint32_t >( OnlineStatus::LookingtoMeldMateria );
-   uint64_t ptMask = uint64_t( 1 ) << static_cast< uint32_t >( OnlineStatus::LookingforParty );
-   uint64_t rpMask = uint64_t( 1 ) << static_cast< uint32_t >( OnlineStatus::Roleplaying );
-
-   OnlineStatus status = OnlineStatus::Online;
-
-   //if( hasStateFlag( Common::PlayerStateFlag::NewAdventurer ) )
-   if( m_onlineStatus & newMask )
-      status = OnlineStatus::NewAdventurer;
-
-   if( m_onlineStatus & afkMask )
-      status = OnlineStatus::AwayfromKeyboard;
-
-   if( m_onlineStatus & busyMask )
-      status = OnlineStatus::Busy;
-
-   if( m_onlineStatus & dcMask )
-      status = OnlineStatus::Disconnected;
+   auto pExdData = g_fw.get< Data::ExdDataGenerated >();
+   if( !pExdData )
+      return OnlineStatus::Online;
 
-   if( m_onlineStatus & meldMask )
-      status = OnlineStatus::LookingtoMeldMateria;
+   uint32_t statusDisplayOrder = 0xFF14;
+   uint32_t applicableStatus = 0;
 
-   if( m_onlineStatus & ptMask )
-      status = OnlineStatus::LookingforParty;
+   for( uint32_t i = 0; i < 64; i++ )
```

Comment:
should not assume a static number here. That is bound to explode.

Reply:
`for( uint32_t i = 0; i < std::numeric_limits< uint64_t >::digits; i++ )` pls